### DISCUSSION TITLE
with chrome cast sent media label and name

### DISCUSSION THREAD

**Discussion by jrousek** [music-assistant/support#596] [Feature requests and ideas]
when I send from Spotify in the music assistant to CromeCast (My TV), only the text (stream from ....) is displayed.

could you please also send information about the current album, the name of the song and image?

So how do you send it from [HA](https://www.home-assistant.io/integrations/cast/)? 

service: media_player.play_media
target:
   entity_id: media_player.chromecast
data:
   media_content_type: "audio/mp3"
   media_content_id: "http://stream.tilos.hu:8000/tilos"
   extras:
     metadata:
       metadataType: 3
       title: "Radio TILOS"
       artist: "LIVE"
       images:
         - url: "https://tilos.hu/images/kockalogo.png"

**Comment by marcelveldt** [comment]
This is not possible as far as I'm aware of without writing a custom media receiver.
The cast device receives the "queue stream" which is a long stream of audio encoded to a format of choice (defaults to lossless flac) of the tracks you have in your queue. There are no audio tracks played on the cast device .

Google Cast does not support "ICY metadata" so we can't encode the metadata in the stream.
Maybe it is possible with a workaround such as sending one feed with the picture to display (or maybe even an interactive OSD) and have the audio feed at the same time. I don't know and I haven't looked at it properly yet. Maybe there are possibilities because HA can also cast the dashboard to the screen.

**Comment by Peterpc58** [comment]
When I stream, I see in Music Assistant for instance for Radio the logo of the radio station and the name of the music that is playing.
If you can't send this info to the speaker, is there another way to use this info? Perhaps with a helper?
When I stream with Tunein Radio from my phone to a speaker, I see all the info. 
I want to use the info to show on a screen. So if a helper is updated with this info, that would be great!

**Reply by OzGav** [reply]
Please see the technical limitation Marcel described above which is due to the way MA works. There is a rewrite due in the next few weeks so perhaps after that this can be revisited.
